### PR TITLE
SDI-603: ✨ Search by physical marks

### PR DESCRIPTION
**Pull request by petergphillips**
- SDI-603: ✨ Index by physical marks
- SDI-603: ✨ Search by physical marks

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt [hunk 1]
```diff
@@ -138,6 +138,63 @@ class PhysicalDetailService(
 
       minShoeSize?.let { detailQuery.filter(rangeQuery("shoeSize").gte(it)) }
       maxShoeSize?.let { detailQuery.filter(rangeQuery("shoeSize").lte(it)) }
+
+      tattoos?.forEach { tattoo ->
+        detailQuery.filter(
+          boolQuery().mustAll(
+            buildList {
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt:145]: quite like this `buildList` syntax - haven't seen it before

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt [hunk 2]
```diff
@@ -138,6 +138,63 @@ class PhysicalDetailService(
...
+              }
+              tattoo.comment?.let {
+                add(QueryBuilders.matchQuery("tattoos.comment", tattoo.comment))
+              }
+            }
+          )
+        )
+      }
+      scars?.forEach { scar ->
+        detailQuery.filter(
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt:157]: bit of duplication here with the queries - will sort out in the next PR as need to work out what should be `must` and what should be `should`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/PhysicalDetailRequest.kt [hunk 3]
```diff
@@ -5,7 +5,7 @@ import io.swagger.v3.oas.annotations.media.Schema
 data class PhysicalDetailRequest(
   @Schema(
     description = "List of prison codes to filter results by",
-    example = "['LEI', 'MDI']",
+    example = "['LEI', 'MDI'],",
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/PhysicalDetailRequest.kt:8]: Rogue comma?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/PhysicalDetailRequest.kt [hunk 4]
```diff
@@ -111,9 +111,42 @@ data class PhysicalDetailRequest(
...
+    example = "Finger",
+    allowableValues = ["Ankle", "Arm", "Ear", "Elbow", "Face", "Finger", "Foot", "Hand", "Head", "Knee", "Leg", "Lip", "Neck", "Nose", "Shoulder", "Thigh", "Toe", "Torso"],
+  )
+  val tattoos: List<BodyPart>? = null,
+
+  @Schema(
+    description = "List of body parts that have marks, searching on the description in the type BODY_PART in the REFERENCE_CODES table. Allowable values extracted 08/02/2023.",
+    example = "Elbow",
+    allowableValues = ["Ankle", "Arm", "Ear", "Elbow", "Face", "Finger", "Foot", "Hand", "Head", "Knee", "Leg", "Lip", "Neck", "Nose", "Shoulder", "Thigh", "Toe", "Torso"],
+  )
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/PhysicalDetailRequest.kt:125]: Are the body parts stored in Nomis somewhere? Might be better to point there?

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**